tools/snippet-testing/model/dependency.rb

```diff
@@ -10,33 +12,94 @@ class Dependency
     RUBY_NAME       = 'ruby'.freeze
     NODE_NAME       = 'node'.freeze
 
-    CS_V4 = '4.7.2'
-    CS_V5 = '5.5.0'
-
     AVAILABLE_LIBRARY_VERSION = {
-      CSHARP_NAME => ['4.x', '5.x'],
+      CSHARP_NAME => ['4.7.2', '5.x'],
       PHP_NAME    => ['4.10', '5.9.0'],
       PYTHON_NAME => ['5.6.0', '6.2.0'],
       RUBY_NAME   => ['4.13.0', '5.0.0.rc20'],
       NODE_NAME   => ['2.11.0', '3.0.0']
     }.freeze
 
-    CSHARP_DEPENDENCIES = {
-      AVAILABLE_LIBRARY_VERSION[CSHARP_NAME][0] => [
-        { name: 'Twilio', version: CS_V4 },
+    CSHARP_DEPENDENCIES = {}
+
+    def initialize
+      update_library_versions
+    end
+
+    def latest_version(versions)
+      versions = VersionSorter.sort(versions)
+      rc_or_alpha = /alpha1$|alpha\-1$|[0-9]{1,}a1$|rc[0-9]+$/
+      last_is_unstable = versions[-1] =~ rc_or_alpha
+
+      if versions[-2] =~ rc_or_alpha && !last_is_unstable
+        versions[-2]
+      else
+        versions[-1]
+      end
+    end
+
+    def csharp_latest
+      response = Unirest.get 'https://api.nuget.org/v3/registration1/twilio/index.json'
+      versions = response.body['items'].map { |version_grp| version_grp['upper'] }
+      latest_version(versions)
+    end
+
+    def php_latest
+      response = Unirest.get 'https://packagist.org/p/twilio/sdk.json'
+      package_info = response.body['packages']['twilio/sdk']
+      latest_version(package_info.keys)
+    end
+
+    def python_latest
+      response = Unirest.get 'https://pypi.python.org/pypi/twilio/json'
+      package_info = response.body['releases']
+      latest_version(package_info.keys)
+    end
+
+    def ruby_latest
+      response = Unirest.get 'https://rubygems.org/api/v1/versions/twilio-ruby.json'
+      versions = response.body.map { |version_info| version_info['number'] }
+      latest_version(versions)
+    end
+
+    def node_latest
+      response = Unirest.get 'http://registry.npmjs.org/-/package/twilio/dist-tags'
+      response.body['alpha']
+    end
+
+    def update_latest_version
+      AVAILABLE_LIBRARY_VERSION[CSHARP_NAME][1] = csharp_latest
+      AVAILABLE_LIBRARY_VERSION[PHP_NAME][1] = php_latest
+      AVAILABLE_LIBRARY_VERSION[PYTHON_NAME][1] = python_latest
+      AVAILABLE_LIBRARY_VERSION[RUBY_NAME][1] = ruby_latest
+      AVAILABLE_LIBRARY_VERSION[NODE_NAME][1] = node_latest
+
+      puts AVAILABLE_LIBRARY_VERSION
+    end
+
+    def update_library_versions
+      update_latest_version
```

Comment:
This function is being called in two different places, the constructor, and this function. The constructor guarantees the function will always be called so there will be no need to call it again later in the same run. The previous code was at least caching the version numbers but now we are requesting twice. Can we remove this call here WDYT?